torch/csrc/autograd/functions/tensor.cpp

```diff
@@ -20,7 +20,8 @@ auto CopyBackwards::apply(const variable_list& grads) -> variable_list {
     grad_inputs[0] = at::zeros_like(grad);
```

Comment:
In general I don't think we guarantee in what context will the function execute, so every time you create a tensor in autograd, you should ensure that there's `AutoGPU` guarding the context. This `zeros_like` might end up in a different place than you want

Reply:
this code is all over the place in the generated code (it's just noticeable here because we don't generate the binding code for copy).  @colesbury do we need to deal with this somehow in the backwards?

Reply:
So the reason why it works for most functions is that they only operate on a single device, and in backwards will be executed in the thread that's pinned to that device (and has it constantly selected in its context). This copy implementation should work as well because we use the first input to dispatch the op, but that's an implementation detail that we shouldn't depend on IMO.

Reply:
Let's not stick AutoGPU calls everywhere. We're always on the device of `grads[0]`

Reply:
I’m fine with this, but let’s add that as a comment in the place where we select the queues in engine. I had a change in mind that would improve the interactions with cpu power saving, but it would sometimes break that invariant.

Reply:
Also, this is currently not true in the JIT interpreter